src/api.js

```diff
@@ -229,6 +238,74 @@ class DICOMwebClient {
         }
       }
 
+      // Event triggered while upload progresses
+      if ('uploadCallbacks' in options) {
+        const {uploadCallbacks} = options;
+
+        if ('loadStart' in uploadCallbacks && typeof uploadCallbacks.loadStart === 'function') {
```

Comment:
I think having this in a loop, where the key is the name of the upload event would make things easy, then just:
if( options?.uploadCallbacks ) {
const { uploadCallbacks } = options;
for(const key of uploadCallbackNames) {
   if( typesof uploadCallbacks[key] === 'function' ) {
      ... add or remove
   }
}
The dicomweb-client is fairly old code style, and the ?. is easier than the 'xxx' in Y && typeof ...

Reply:
this makes sense a lot